pylint/checkers/stdlib.py

```diff
@@ -571,6 +585,29 @@ def visit_boolop(self, node: nodes.BoolOp) -> None:
         for value in node.values:
             self._check_datetime(value)
 
+    @utils.check_messages("lru-cache-decorating-method")
+    def visit_functiondef(self, node: nodes.FunctionDef) -> None:
+        if node.decorators and isinstance(node.parent, nodes.ClassDef):
+            self._check_lru_cache_decorators(node.decorators)
+
+    def _check_lru_cache_decorators(self, decorators: nodes.Decorators) -> None:
+        """Check if instance methods are decorated with functools.lru_cache."""
+        lru_cache_nodes: List[nodes.NodeNG] = []
+        for d_node in decorators.nodes:
+            for infered_node in d_node.infer():
```

Comment:
Shouldn't we use ``safe_infer`` or catch InferenceError here ?

Reply:
Yes we should.. See tests 😅

Reply:
I didn't use `safe_infer` as then we only get one value or `Uninferable` for `lru_cache()` decorators.
I did add a `try...except` though.